.github/workflows/build-ffmpeg.yml

```diff
@@ -24,102 +61,153 @@ jobs:
         run: |
-          cd ~/ffmpeg_sources
-          git -C x264 pull 2> /dev/null || git clone --depth 1 https://code.videolan.org/videolan/x264.git
+          git clone --depth 1 https://code.videolan.org/videolan/x264.git
           cd x264
-          PATH="$HOME/bin:$PATH" PKG_CONFIG_PATH="$HOME/ffmpeg_build/lib/pkgconfig" ./configure \
-            --prefix="$HOME/ffmpeg_build" \
-            --bindir="$HOME/bin" \
+          PATH="$root_path/bin:$PATH" \
+            PKG_CONFIG_PATH="$root_path/ffmpeg_build/lib/pkgconfig" \
+            ./configure \
+            --prefix="$root_path/ffmpeg_build" \
+            --bindir="$root_path/bin" \
             --disable-cli \
             --enable-shared
-          PATH="$HOME/bin:$PATH" make -j$(nproc)
+          PATH="$root_path/bin:$PATH" make -j$(nproc)
           make install
 
       - name: libx265
-        # do we need to build this or can we just install it? https://packages.ubuntu.com/kinetic/libx265-dev
+        env:
+          root_path: ${{ steps.root.outputs.root_path }}
+        working-directory: ffmpeg_sources
         run: |
-          cd ~/ffmpeg_sources
-          wget -O x265.tar.bz2 https://bitbucket.org/multicoreware/x265_git/get/master.tar.bz2
-          tar xjvf x265.tar.bz2
-          cd multicoreware*/build/linux
-          PATH="$HOME/bin:$PATH" cmake -G "Unix Makefiles" \
-            -DCMAKE_INSTALL_PREFIX="$HOME/ffmpeg_build" \
+          git clone https://bitbucket.org/multicoreware/x265_git.git
+          cd x265_git/build/msys
+          PATH="$root_path/bin:$PATH" cmake -G "${{ matrix.cmake_generator }} Makefiles" \
+            -DCMAKE_INSTALL_PREFIX="$root_path/ffmpeg_build" \
             -DENABLE_CLI=off \
             -DENABLE_SHARED=on \
             ../../source
-          PATH="$HOME/bin:$PATH" make -j$(nproc)
+          PATH="$root_path/bin:$PATH" make -j$(nproc)
           make install
 
       - name: nvenc
-        # do we need to build this or can we just install it? https://packages.ubuntu.com/jammy/libffmpeg-nvenc-dev
+        id: nvenc
+        working-directory: ffmpeg_sources
         # https://trac.ffmpeg.org/wiki/HWAccelIntro#NVENC
         run: |
           git clone https://git.videolan.org/git/ffmpeg/nv-codec-headers.git
           cd nv-codec-headers
           make -j$(nproc)
-          sudo make install
+          if [[ ${{ matrix.os_type }} == "windows" ]]; then
+            make install
+          else
+            sudo make install
+          fi
+
+          echo "nvenc_path=$PWD" >> $GITHUB_OUTPUT
 
       - name: ffmpeg
+        env:
+          nvenc_path: ${{ steps.nvenc.outputs.nvenc_path }}
+          root_path: ${{ steps.root.outputs.root_path }}
+        working-directory: ffmpeg_sources
         run: |
-          cd ~/ffmpeg_sources
-          wget -O ffmpeg-snapshot.tar.bz2 https://ffmpeg.org/releases/ffmpeg-snapshot.tar.bz2
-          tar xjvf ffmpeg-snapshot.tar.bz2
+          git clone https://git.ffmpeg.org/ffmpeg.git
```

Comment:
Agree with this. It may be a good idea, at least in the initial PR test stage, to target `release/4.4`. That will minimize random build failures and maximize compatibility with the existing Sunshine source. Once a stable build is proven to work, thought can be given to ffmpeg5.

Reply:
I will pin ffmpeg (and possibly other dependencies) using submodules.